interfaces/adfsuite/ams.py

```diff
@@ -2193,15 +2237,20 @@ class AMSJob(SingleJob):
     """A class representing a single computation with AMS driver. The corresponding results type is |AMSResults|."""
 
     results: AMSResults
-    molecule: Union[Molecule, Dict[str, Molecule], None]
+    molecule: Optional[Union[Molecule, Dict[str, Molecule], "ChemicalSystem"]]
 
     _result_type = AMSResults
     _command = "ams"
 
-    def __init__(self, molecule: Union[Molecule, Dict[str, Molecule], None] = None, *args, **kwargs):
+    def __init__(
+        self, molecule: Optional[Union[Molecule, Dict[str, Molecule], "ChemicalSystem"]] = None, *args, **kwargs
+    ):
+        molecule = molecule.copy() if _has_scm_chemsys and isinstance(molecule, ChemicalSystem) else molecule
```

Comment:
Same thing here: I think a ``Dict[str, "ChemicalSystem"]`` should be fine too? I'm actually wondering about the copying in case a dictionary is passed. Intuitively, I'd expect the copying to happen for all systems in the dict then, but that is not what we are currently doing.

Reply:
Updated to now copy all the elements if a dict is passed in